### PR TITLE
[ENG-71] Add deployments to release workflow

### PR DESCRIPTION
**Pull request by trmartin4**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [X] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
In order to track which Jira issues are included in each deployment, we need to tie the release pipeline to Github deployments, which the Github for Jira integration will then tie to Jira deployments.

## Code changes

* **[release.yml](https://github.com/bitwarden/directory-connector/compare/ENG-71-add-deployment?expand=1#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34)** Added workflow steps to create a Github deployment and update the status to success or failure based on the job success or failure.

## Before you submit
- [ ] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [X] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -60,6 +60,23 @@ jobs:
...
           echo "::set-output name=branch-name::$BRANCH_NAME"
 
+      - name: Create GitHub deployment
+        uses: chrnorm/deployment-action@1b599fe41a0ef1f95191e7f2eec4743f2d7dfc48
+        id: deployment
+        with:
+          token: '${{ secrets.GITHUB_TOKEN }}'
+          environment: 'Directory Connector - Production'
+          description: 'Deployment ${{ steps.retrieve-version.outputs.package_version }} from branch ${{ branch.branch-name }}'
+          task: release
```
Comment (vgrassia) [inline .github/workflows/release.yml:70]:
```suggestion
          token: '${{ secrets.GITHUB_TOKEN }}'
          initial-status: 'in_progress'
          environment: 'production'
          description: 'Deployment ${{ steps.retrieve-version.outputs.package_version }} from branch ${{ steps.branch.outputs.branch-name }}'
          task: release
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -60,6 +60,23 @@ jobs:
...
+      
+      - name: Update deployment status to In Progress
+        uses: chrnorm/deployment-status@07b3930847f65e71c9c6802ff5a402f6dfb46b86
+        with:
+          token: '${{ secrets.GITHUB_TOKEN }}'
+          state: 'in_progress'
+          deployment-id: ${{ steps.deployment.outputs.deployment_id }}
```
Comment (vgrassia) [inline .github/workflows/release.yml:77]:
```suggestion
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -104,3 +121,19 @@ jobs:
           body: "<insert release notes here>"
           token: ${{ secrets.GITHUB_TOKEN }}
           draft: true
+          
+      - name: Update deployment status to Success
+        if: success()
```
Comment (vgrassia) [inline .github/workflows/release.yml:126]:
```suggestion
        if: ${{ success() }}
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -104,3 +121,19 @@ jobs:
...
+        uses: chrnorm/deployment-status@07b3930847f65e71c9c6802ff5a402f6dfb46b86
+        with:
+          token: '${{ secrets.GITHUB_TOKEN }}'
+          state: 'success'
+          deployment-id: ${{ steps.deployment.outputs.deployment_id }}
+
+      - name: Update deployment status to Failure
+        if: failure()
```
Comment (vgrassia) [inline .github/workflows/release.yml:134]:
```suggestion
        ${{ if: failure() }}
```

**Review by vgrassia — CHANGES_REQUESTED** [review]
The `setup` job also needs the following [permissions](https://github.com/chrnorm/deployment-action#example-usage).
```
permissions:
  deployments: write
```

File: .github/workflows/release.yml (hunk 4 above)
Comment (trmartin4) [reply .github/workflows/release.yml]: I adjusted this to be `if: ${{ failure() }}` as I believe it's supposed to be and commited that.

**Review by vgrassia — APPROVED** [review]

**Pull request merged**